harfbuzz-traits/src/unicode_funcs_traits.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright 2023 The Servo Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+/// A general category value. Equivalent to [`hb_unicode_general_category_t`].
+///
+/// [`hb_unicode_general_category_t`]: crate::sys::hb_unicode_general_category_t
+#[repr(u8)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+#[allow(non_camel_case_types)] // the names are defined by Unicode
+#[allow(missing_docs)] // the categories are defined by Unicode
+pub enum GeneralCategory {
+    Control = 0,
+    Format = 1,
+    Unassigned = 2,
+    Private_use = 3,
```

Comment:
`use` should be `Use` with upper-case `U`.

I still think the underscores in these values should be removed the way ICU4X removes the upstream Unicode underscores to get idiomatic Rust enum values.

Reply:
I agree with both of those requests, fwiw.